blockstore/apps/api/v1/views/snapshots.py

```diff
@@ -0,0 +1,109 @@
+"""
+Views for Snapshots.
+"""
+
+from django import http
+from rest_framework import status
+from rest_framework import viewsets
+from rest_framework.response import Response
+
+from blockstore.apps.bundles.store import BundleDataStore, BundleSnapshot
+from blockstore.apps.bundles.models import BundleVersion
+
+from ...constants import FILE_PATH_REGEX
+from ..serializers.snapshots import FileInfoSerializer
+
+
+class BundleFileReadOnlyViewSet(viewsets.ViewSet):
+    """
+    Read only ViewSet for files in a BundleVersion.
+    """
+
+    detail_view_name = 'api:v1:bundleversionfile-detail'
+
+    lookup_field = 'path'
+    lookup_value_regex = FILE_PATH_REGEX
+
+    serializer_class = FileInfoSerializer
+
+    def get_serializer_context(self):
+        return {
+            'detail_view_name': self.detail_view_name,
+            'format': self.format_kwarg,
+            'request': self.request,
+        }
+
+    def get_bundle_version(self, bundle_uuid, version_num=None):
+
+        filter_kwargs = {
+            'bundle__uuid': bundle_uuid
+        }
+        if version_num:
+            filter_kwargs['version_num'] = version_num
+
+        try:
+            return BundleVersion.objects.filter(**filter_kwargs).order_by('-version_num').first()
+        except BundleVersion.DoesNotExist:
+            pass
+        return None
+
+    def get_bundle_version_or_404(self, bundle_uuid, version_num=None):
+        bundle_version = self.get_bundle_version(bundle_uuid, version_num)
+        if bundle_version:
+            return bundle_version
+        raise http.Http404
+
+    def list(self, _request, bundle_uuid, version_num=None):
+        snapshot = self.get_bundle_version_or_404(bundle_uuid, version_num).snapshot()
+        file_info_serializer = FileInfoSerializer(
+            snapshot.files.values(),
+            context=self.get_serializer_context(),
+            many=True
+        )
+        return Response(file_info_serializer.data)
+
+    def retrieve(self, _request, bundle_uuid, path, version_num=None):
+        snapshot = self.get_bundle_version_or_404(bundle_uuid, version_num).snapshot()
+        file = snapshot.files.get(path, None)
+        if file:
+            file_info_serializer = FileInfoSerializer(
+                file,
+                context=self.get_serializer_context(),
+            )
+            return Response(file_info_serializer.data)
+        raise http.Http404
+
+
+class BundleFileViewSet(BundleFileReadOnlyViewSet):
+    """
+    ViewSet for files in the most recent BundleVersion of a Bundle.
+    """
+
+    detail_view_name = 'api:v1:bundlefile-detail'
+
+    def create(self, request, bundle_uuid, version_num=None):
```

Comment:
@UmanShahzad I'm having trouble posting multiple files to this endpoint -- is there a way to make this work?  Or as an alternative, can you enable PUT requests to this endpoint which appends files to the bundle?

See [openedx-blockstore-relay#1](https://github.com/open-craft/openedx-blockstore-relay/pull/1/files#diff-e44130bcee7479b4f9a12331518df666R98) for my WIP attempt to post multiple files in a single request.

Reply:
@pomegranited Since each file is considered a resource, right now this will require posting each file independently. I have added a point to LX-53 to investigate how we can add support for posting multiple files at once.

P.S. HTTP 400 Bad Request: The tagged Github id is not a participant of this PR. ;)

Reply:
HTTP 501: I cannot fulfill this request

Reply:
Sorry for the mis-tag -- I blame tab-complete 😄 

I'm cool with posting multiple file upload requests, but that's not working with the current API.  A POST replaces all the files in a bundle, and PUTs aren't allowed.  Can you allow PUTs here?

Reply:
That is weird. Can you check what requests the DRF web interface is sending? Because I can add multiple files via that. And there is a test for this too: https://github.com/open-craft/blockstore/pull/4/files#diff-3959f0e2e005ddd8ad570fb6febdea94R332. Since the view logic is custom, merely changing the HTTP verb should not change the behavior.

Reply:
Ah oops..  you're totally right @symbolist .  I thought that POST would replace all the files, but it doesn't, it's just additive.  If I want replace them all, I'll need to iterate over the undesired files and DELETE them.

Thanks for your help!